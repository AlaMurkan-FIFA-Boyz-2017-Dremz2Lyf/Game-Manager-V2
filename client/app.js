// Dependancies
import React from 'react';
import ReactDOM from 'react-dom';
import { ProviderÂ } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import promise from 'redux-promise';

// Components
import Home from './components/home';

import rootReducer from './reducers/index.js';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(rootReducer, undefined, composeEnhancers(
  applyMiddleware(promise)
));

ReactDOM.render(
  <Provider store={store}>
    <Home/>
  </Provider>,
  document.getElementById('app')
);
