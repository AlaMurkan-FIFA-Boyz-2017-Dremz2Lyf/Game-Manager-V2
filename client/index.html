<html>
<head>
  <title>Game Manager</title>
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans|Oswald|PT+Sans">
</head>
<body>
  <div id="app"></div>
</body>
</html>
