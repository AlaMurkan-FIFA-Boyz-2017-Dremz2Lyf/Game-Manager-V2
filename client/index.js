// Dependancies
import React from 'react';
import ReactDOM from 'react-dom';
import { ProviderÂ } from 'react-redux';

// css
import '../styles/styles.css';

// Bring in our router
import Routes from './router';

// Grab our root reducer and our store from their files
import rootReducer from './reducers/index.js';
import { store } from './store';

// Found this... Needed for hot module reloading.
if (module.hot) {
  module.hot.accept();
}

ReactDOM.render(
  <Provider store={store}>
    <Routes />
  </Provider>,
  document.getElementById('app')
);
