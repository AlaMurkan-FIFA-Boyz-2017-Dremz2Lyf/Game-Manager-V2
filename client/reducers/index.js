import { combineReducers } from 'redux';
import data from './data_reducer';
import isError from './isError';
import isLoading from './isLoading';

const rootReducer = combineReducers({
  data,
  isError,
  isLoading
});

export default rootReducer;
