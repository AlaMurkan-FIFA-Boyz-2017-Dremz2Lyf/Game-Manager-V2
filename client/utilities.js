/**
  * normalize
  * @param {array} list - an array of objects to be normalized into an object
*/
export const normalize = (list) => {

  return list.reduce((normalized, item) => {
    normalized[item.id] = item;
    return normalized;
  }, {});

};

/**
  * applyAll - apply all uses a bit of tricky programming here to return a new state adding to only the specific substate we need.
  * @param {object} state - the current state
  * @param {object} action - the action coming through, it's payload must have a stateKey and the data corresponding to that statekey.
*/
export const applyAll = (state, action) => {
  const { stateKey, data } = action.payload;

  return {...state, [stateKey]: data};
};

/**
  * percentPlayed - basic math here.
  * @param {number} gamesPlayed - pretty self explanatory
  * @param {number} totalGames - pretty self explanatory
*/
export const percentPlayed = (gamesPlayed, totalGames) => {
  return Math.floor((gamesPlayed / totalGames) * 100);
};

/**
  * getValidationState - will give us the correct bootstrap class for UI responsiveness
  * @param {bool} valid - passed from redux-form, lets us know if the form is valid
  * @param {bool} pristine - from redux-form, lets us know if the form has been touched & edited
*/
export const getValidationState = (valid, pristine) => pristine ? null : (valid ? 'success' : 'error' );

/**
  * required - another validation check for redux-form
  * @param {string} value - from redux-form, returns an alert if the user has not entered required info. Also deals with the number 0 as that is falsey value. 
*/
export const required = value => value === 0 || value ? undefined : 'Required';

/**
  * possValidation - validation specific to the possession field
  * @param {string} value - the input from the form
  * @param {string} allValues - all the inputs from the form
*/
export const possValidation = (value, allValues) => {
  const { p1Poss, p2Poss } = allValues;

  if (!p1Poss && !p2Poss) {
    return 'Possession is required.';
  } else if (Number(p1Poss) + Number(p2Poss) !== 100) {
    return 'Possession should add up to 100%!';
  }
  return undefined;
};

/**
  * helpMessage - renders the correct help message based on the field
  * @param {string} field
*/
export const helpMessage = (field) => {
  let message;

  if (field === 'Score') {
    message = `${field} is required! I mean.. come on!`;
  } else {
    message = `${field} will default to 0 if not given.`;
  }
  return message;
};

/**
  * buildTable
  * @param {object} games - a normalized games object to build the table from
  * @param {object} players - another normalized object to reference the player names.
*/
export const buildTable = (games, players) => {

  let trackedStats = {
    wins: 0, losses: 0, draws: 0, goalsFor: 0, goalsAgainst: 0, shots: 0, onGoal: 0, reds: 0, yellows: 0, poss: null, points: 0, goalDiff: 0
  };

  return Object.values(games).reduce((table, game) => {

    let { p1, p2 } = game;

    table[p1] = table[p1] ? table[p1] : { ...trackedStats, id: p1, username: players[p1].username};
    table[p2] = table[p2] ? table[p2] : { ...trackedStats, id: p2, username: players[p2].username};

    return applyGame(table, game);
  }, {});
};


// I extracted this out of the reduce in build table because of all the action that needs to happen.
// There might be a more programmatic way to do this. Potentially mapping the game objects from the database into nested objects for each player in the game.... that would, in theory, allow us to just loop through each player's object from the game and apply them to the correct player.

/**
  * applyGame - used in buildTable to apply the game to the table
  * @param {object} table - the current table
  * @param {object} game - the current game to be applied
*/
export const applyGame = (table, game) => {

  let { p1, p2, p1Score, p2Score, p1Shots, p2Shots, p1Poss, p2Poss, p1OnGoal, p2OnGoal, p1Reds, p1Yellows, p2Reds, p2Yellows } = game;


  let player1 = table[p1];
  let player2 = table[p2];

  if (p1Score && p2Score) {

    p1Score === p2Score ? (
      player1.draws++,
      player2.draws++
    ) : p1Score > p2Score ? (
      player1.wins++,
      player2.losses++
    ) : (
      player2.wins++,
      player1.losses++
    );
    player1.goalsFor += p1Score;
    player2.goalsFor += p2Score;
    player1.goalsAgainst += p2Score;
    player2.goalsAgainst += p1Score;
    player1.shots += p1Shots;
    player2.shots += p2Shots;
    player1.onGoal += p1OnGoal;
    player2.onGoal += p2OnGoal;
    player1.reds += p1Reds;
    player2.reds += p2Reds;
    player1.yellows += p1Yellows;
    player2.yellows += p2Yellows;
    player1.poss = player1.poss === null ? p1Poss : (player1.poss + p1Poss) / (player1.wins + player1.losses + player1.draws);
    player2.poss = player2.poss === null ? p2Poss : (player1.poss + p1Poss) / (player1.wins + player1.losses + player1.draws);
    player1.points = (player1.wins * 3 + player1.draws);
    player2.points = (player2.wins * 3 + player2.draws);
    player1.goalDiff = (player1.goalsFor - player1.goalsAgainst);
    player2.goalDiff = (player2.goalsFor - player2.goalsAgainst);
  }
  return table;
};


// NOTE: Currently not being used:
  // Need to figure out how to get the react-bootstrap - tooltips to work.
  // Seemingly I need to build a component wrapped in a <th>... but the first attempt didn't work.
export const mapHeaders = (header) => {
  switch (header) {
  case 'P':
    return 'Points';
  case 'W':
    return 'Wins';
  case 'L':
    return 'Losses';
  case 'D':
    return 'Draws';
  case 'GF':
    return 'Goals For';
  case 'GA':
    return 'Goals Against';
  case 'GD':
    return 'Goal Differencial';
  case 'S':
    return 'Shots';
  case 'OT':
    return 'Shots On Target';
  case 'Y':
    return 'Yellows';
  case 'R':
    return 'Reds';
  case 'Po':
    return 'Possession';
  default:
    return 'Player';
  }
};
