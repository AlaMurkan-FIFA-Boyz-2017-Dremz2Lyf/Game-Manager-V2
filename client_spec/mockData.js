// NOTE: Axios is required here to pass it to the mock adapter function.
const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

//Mock Data for our axios mocks
export const mockData = {};

// Create a fake date to standardize createdAt and updatedAt values through tests
mockData.createdAt = '2016-12-25T19:31:48';
mockData.updatedAt = '2016-12-26T19:31:48';


mockData.test = {
  existing: [
    {
      id: 1,
      laborum: 'officia proident duis in commodo',
      veniam: 'irure irure amet aliquip consectetur.'
    },
    {
      id: 2,
      laborum: 'Ea est eiusmod laborum elit ut amet id',
      veniam: 'Esse minim irure eu occaecat veniam duis.'
    },
    {
      id: 3,
      laborum: 'Ea est officia laborum elit ut amet id',
      veniam: 'Duis irure eu laborum amet duis.'
    }
  ],
  updated: [
    {
      id: 1,
      laborum: 'Updated info',
      veniam: 'Ya know it'
    }
  ]
};

mockData.table = {
  1: {id: 1, username: 'Alice', points: 3, wins: 1, losses: 1, draws: 0, goalsFor: 4, goalsAgainst: 4, goalDiff: 0, shots: 0, onGoal: 0, reds: 0, yellows: 0, poss: 62.5},
  2: {id: 2, username: 'Gilbert', points: 0, wins: 0, losses: 1, draws: 0, goalsFor: 2, goalsAgainst: 3, goalDiff: -1, shots: 0, onGoal: 0, reds: 0, yellows: 0, poss: 25},
  3: {id: 3, username: 'Bob', points: 3, wins: 1, losses: 0, draws: 0, goalsFor: 2, goalsAgainst: 1, goalDiff: 1, shots: 0, onGoal: 0, reds: 0, yellows: 0, poss: 50}
};


// set a players key on the mock database
mockData.players = [
  {id: 1, username: 'Alice', wins: 0, losses: 0, draws: 0, goalsFor: 0, goalsAgainst: 0, shots: 0, onGoal: 0, reds: 0, yellows: 0, possession: 0, isTeam: 0, createdAt: '2016-12-25T19:31:48', updatedAt: null},
  {id: 2, username: 'Gilbert', wins: 0, losses: 0, draws: 0, goalsFor: 0, goalsAgainst: 0, shots: 0, onGoal: 0, reds: 0, yellows: 0, possession: 0, isTeam: 0, createdAt: '2016-12-25T19:31:48', updatedAt: null},
  {id: 3, username: 'Bob', wins: 0, losses: 0, draws: 0, goalsFor: 0, goalsAgainst: 0, shots: 0, onGoal: 0, reds: 0, yellows: 0, possession: 0, isTeam: 0, createdAt: '2016-12-25T19:31:48', updatedAt: null}
];

// set a tournaments key on the mock database
mockData.tournaments = [
  {id: 1, gamesPlayed: 0, totalGames: 3, name: 'Super Tourney!', winner: null, createdAt: '2016-12-25T19:31:48', updatedAt: null},
  {id: 2, gamesPlayed: 1, totalGames: 1, name: 'new', winner: 2, createdAt: '2016-12-25T19:31:48', updatedAt: mockData.updatedAt},
  {id: 3, gamesPlayed: 5, totalGames: 5, name: 'next', winner: null, createdAt: '2016-12-25T19:31:48', updatedAt: null},
  {id: 4, gamesPlayed: 1, totalGames: 7, name: 'hot', winner: null, createdAt: '2016-12-25T19:31:48', updatedAt: null},
  {id: 5, gamesPlayed: 2, totalGames: 6, name: 'fun', winner: null, createdAt: '2016-12-25T19:31:48', updatedAt: null}
];
// set a games key on the mock database
mockData.games = [
  {id: 1, p1: 1, p2: 2, p1Score: null, p2Score: null, p1Shots: null, p2Shots: null, p1Poss: null,
    p2Poss: null, p1OnGoal: null, p2OnGoal: null, tournament: 1, status: 'created', p1Reds: 0, p1Yellows: 0, p2Reds: 0, p2Yellows: 0, createdAt: mockData.createdAt, updatedAt: null},
  {id: 2, p1: 1, p2: 3, p1Score: null, p2Score: null, p1Shots: null, p2Shots: null, p1Poss: null,
    p2Poss: null, p1OnGoal: null, p2OnGoal: null, tournament: 1, status: 'created', p1Reds: 0, p1Yellows: 0, p2Reds: 0, p2Yellows: 0, createdAt: mockData.createdAt, updatedAt: null},
  {id: 3, p1: 2, p2: 3, p1Score: null, p2Score: null, p1Shots: null, p2Shots: null, p1Poss: null,
    p2Poss: null, p1OnGoal: null, p2OnGoal: null, tournament: 1, status: 'created', p1Reds: 0, p1Yellows: 0, p2Reds: 0, p2Yellows: 0, createdAt: mockData.createdAt, updatedAt: null},
  {id: 4, p1: 2, p2: 3, p1Score: 2, p2Score: 1, p1Shots: 5, p2Shots: 6, p1Poss: 50,
    p2Poss: 50, p1OnGoal: 1, p2OnGoal: 3, tournament: 2, status: 'finished', p1Reds: 0, p1Yellows: 0, p2Reds: 0, p2Yellows: 0, createdAt: mockData.createdAt, updatedAt: mockData.updatedAt}
];

mockData.newGame = {
  id: 4, p1: 2, p2: 3, p1Score: null, p2Score: null, p1Shots: null, p2Shots: null, p1Poss: null,
  p2Poss: null, p1OnGoal: null, p2OnGoal: null, tournament: 2, status: 'created', p1Reds: 0, p1Yellows: 0, p2Reds: 0, p2Yellows: 0, createdAt: mockData.createdAt, updatedAt: null};



mockData.updated;

// This lets us test our axios requests easily.
export const mockAxios = new MockAdapter(axios);

// setup our routes and methods
mockAxios.onGet('/games').reply((config) => {
  if (config.params.type && config.params.type === 'tournament') {
    return [200, mockData.games.filter(game => game.tournament === config.params.id)];
  } else {
    return config.params.id ? (
      config.params.id === 4 ? [200, [mockData.newGame]] : [200, [mockData.updated]]) : ([200, mockData.games]);
  }
})
.onPost('/games').reply(() => [202, [mockData.newGame]])
.onPut('/games').reply((config) => {
  mockData.updated = JSON.parse(config.data);

  return [201, [mockData.updated]];
});

mockAxios.onGet('/tournaments').reply((config) => {
  if (config.params.rando === 'somethingInvalid') {
    return [400, 'Bad request'];
  }
})
.onPost('/tournaments').reply((config) => {
  if (config.data === 'somethingInvalid') {
    return [400, 'Bad request'];
  }
})
.onPut('/tournaments').reply((config) => {
  if (config.data === 'somethingInvalid') {
    return [400, 'Bad request'];
  }
});
