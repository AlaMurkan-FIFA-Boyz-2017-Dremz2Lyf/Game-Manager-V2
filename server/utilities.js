/*
  createGames is a function that accepts two arguments
    - players: a list of all the players to be created <Array>
    - tournament: a number representing the tournament the games will be affiliated with. <Number>
*/
exports.createGames = (players, tournament, rounds = 1) => {

  // Handle validation for the first argument here.
  if (!Array.isArray(players)) {
    throw new TypeError('createGames\' first argument must be an array.', 'utilities.js', 6);
  }

  // Handle validation for the second argument next.
  if (typeof tournament !== 'number') {
    throw new TypeError('createGames\' second argument must be a number', 'utilities.js', 6);
  }

  // Then make a copy of the input array.
  let list = players.slice();

  // Create a games array.
  let games = [];

  // While there is something in the copy of the players list,
  while (list.length > 0) {

    // grab the first player from it,
    let p1 = list.pop();

    // and go over each other player and create a game with those two players.
    list.forEach(p2 => {
      // handle some double check validation here.
      if (typeof p2 !== 'number' || typeof p1 !== 'number') {
        throw new TypeError('Items in the array passed to createGames should be numbers', 'utilities.js', 6);
      } else {
        let game = {p1, p2};

        // Set a tournament on the game and push it into the games array.
        game.tournament = tournament;
        games.push(game);
      }
    });
  }

  // applyRounds and return the resulting games array once it is all finished.

  return exports.applyRounds(games, rounds);
};

exports.formatName = (string) => {
  // Handle validation here
  if (typeof string !== 'string') {
    throw new TypeError('input must be a string', 'utilities.js', 49);
  }
  // then take care of the actual formatting here and return it. Capitalize the first letter of the string, and every other letter to lowercase.
  return string.split('').map((char, index) => (
    index === 0 ? char.toUpperCase() : char.toLowerCase()
  )).join('');
};


// ApplyRounds takes the build games array and creates duplicates for the number of rounds asked.
exports.applyRounds = (gamesArray, rounds) => {
  // combined starts as the input array of games
  let combined = gamesArray;

  // for the number of rounds passed in, add the games array to the combined holder
    // and let combined be that result.
  for (let i = 1; i < rounds; i++) {
    combined = combined.concat(gamesArray);
  }

  // return the result
  return combined;
};

exports.createTable = (players, games) => {

  let trackedStats = {
    wins: 0, losses: 0, draws: 0, goalsFor: 0, goalsAgainst: 0, shots: 0, onGoal: 0, reds: 0, yellows: 0, points: 0, goalDiff: 0,
  };

  let table = games.reduce((table, game) => {

    let { p1, p2 } = game;

    table[p1] = table[p1] ? table[p1] : Object.assign({}, trackedStats, {id: p1}, {poss: [], passAcc: []});
    table[p2] = table[p2] ? table[p2] : Object.assign({}, trackedStats, {id: p2}, {poss: [], passAcc: []});

    return exports.applyGame(table, game);
  }, {});

  players.forEach(player => {
    let { id, username } = player;
    if (table[id]) {
      table[id].username = username;
    }
  });

  return table;
};

exports.applyGame = (table, game) => {

  let { p1, p2, p1Score, p2Score, p1Shots, p2Shots, p1Poss, p2Poss, p1OnGoal, p2OnGoal, p1Reds, p1Yellows, p2Reds, p2Yellows, p1PassAcc, p2PassAcc } = game;

  let player1 = table[p1];
  let player2 = table[p2];

  if (p1Score && p2Score) {

    p1Score === p2Score ? (
      player1.draws++,
      player2.draws++
    ) : p1Score > p2Score ? (
      player1.wins++,
      player2.losses++
    ) : (
      player2.wins++,
      player1.losses++
    );
    player1.goalsFor += p1Score;
    player2.goalsFor += p2Score;
    player1.goalsAgainst += p2Score;
    player2.goalsAgainst += p1Score;
    player1.shots += p1Shots;
    player2.shots += p2Shots;
    player1.onGoal += p1OnGoal;
    player2.onGoal += p2OnGoal;
    player1.reds += p1Reds;
    player2.reds += p2Reds;
    player1.yellows += p1Yellows;
    player2.yellows += p2Yellows;
    player1.poss.push(p1Poss);
    player2.poss.push(p2Poss);
    player1.passAcc.push(p1PassAcc);
    player2.passAcc.push(p2PassAcc);
    player1.points = (player1.wins * 3 + player1.draws);
    player2.points = (player2.wins * 3 + player2.draws);
    player1.goalDiff = (player1.goalsFor - player1.goalsAgainst);
    player2.goalDiff = (player2.goalsFor - player2.goalsAgainst);
  }
  return table;
};

exports.envStaticPath = (env) => (
  env === 'production' ? 'public' : 'devPublic'
);
