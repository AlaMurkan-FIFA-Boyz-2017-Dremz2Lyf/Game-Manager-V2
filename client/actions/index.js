// dependancies
import axios from 'axios';

// Action Types
import { SET_ERRORED, SET_LOADING, RECEIVE } from './types';

// Utilities
import { normalize } from '../utilities';

export const setErrored = (stateKey, data) => {
  // NOTE: Still not 100% sure if this is going to work for all errors.... but it might! :D
  let error = data.status ? {status: data.status, message:Â data.data} : {message: data.message, source: data.stack};

  return {
    type: SET_ERRORED,
    payload: {
      data: error,
      stateKey
    }
  };
};


export const setLoading = (stateKey, data) => {
  return {
    type: SET_LOADING,
    payload: {
      data,
      stateKey
    }
  };
};

export const receive = (stateKey, data) => {
  // receive is set up to accept an array or an object.
    // If it is an array we do the normalization here.
  if (Array.isArray(data)) {
    data = normalize(data);
  }

  // and send off our action with the proper payload
  return {
    type: RECEIVE,
    payload: {
      data,
      stateKey
    }
  };
};

export const fetch = (stateKey, data = {}) => {
  let request = axios.get(`/${stateKey}`, {params: data});

  return dispatch => {
    dispatch(setLoading(stateKey, true));

    return request.then(response => response.data).then(data => {
      dispatch(receive(stateKey, data));
    }).catch(error => {
      // This should handle any response errors, but more than likely not anything else.... D:
      let data = error.response ? error.response : error;
      dispatch(setErrored(stateKey, data));
    }).then(dis => {
      dispatch(setLoading(stateKey, false));
    });
  };
};

// In an effort to minimize request traffic, the server will respond with the updated or created data object we are currently working with. Instead of fetching all of that corresponding data we need after the post. We will just pass the updated data through to the receive action, by using getState from redux-thunk.

export const create = (stateKey, data) => {
  let post = axios.post(`/${stateKey}`, data);

  return (dispatch, getState) => {
    dispatch(setLoading(stateKey, true));

    return post.then(response => normalize(response.data)).then(updated => {

      // Grab the specific subState for the stateKey we are working with during this action
      let subState = getState().data[stateKey];

      // Take the subState and the updated data and create a new object
      let updatedSubState = {...subState, ...updated};

      // Pass that updated subState into the receive action
      return dispatch(receive(stateKey, updatedSubState));
    }).catch(error => {

      // If there was an error in the put, we grab the response from the server and pass it along to our setErrored action creator.
      let data = error.response;
      return dispatch(setErrored(stateKey, data));
    }).then(dis =>

      // Once everything is done, we set our loading flag to false.
      dispatch(setLoading(stateKey, false))
    );
  };
};

export const update = (stateKey, data) => {
  let put = axios.put(`/${stateKey}`, data);

  return (dispatch, getState) => {
    // Set our loading flag for the specific stateKey we are working with
    dispatch(setLoading(stateKey, true));

    // Then handle the async request.
    return put.then(response => normalize(response.data)).then(updated => {

      // Grab the specific subState for the stateKey we are working with during this action
      let subState = getState().data[stateKey];

      // Take the subState and the updated data and create a new object
      let updatedSubState = {...subState, ...updated};

      // Pass that updated subState into the receive action
      return dispatch(receive(stateKey, updatedSubState));
    }).catch(error => {

      // If there was an error in the put, we grab the response from the server and pass it along to our setErrored action creator.

      let data = error.response;
      return dispatch(setErrored(stateKey, data));
    }).then(dis =>

      // Once everything is done, we set our loading flag to false.
      dispatch(setLoading(stateKey, false))
    );
  };
};
