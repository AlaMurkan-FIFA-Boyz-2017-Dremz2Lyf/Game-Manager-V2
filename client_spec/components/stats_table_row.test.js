import React from 'react';
import TableRow from '../../client/components/stats_table_row';

import { buildTable, normalize } from '../../client/utilities';
import { mockData } from '../mockData';


describe('TableRow', () => {
  let { players, games } = mockData;

  players = normalize(players);
  games = normalize(games);
  let table = buildTable(games, players);

  let wrapper = shallow(
    <TableRow
      player={table[1]}
    />
  );

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
