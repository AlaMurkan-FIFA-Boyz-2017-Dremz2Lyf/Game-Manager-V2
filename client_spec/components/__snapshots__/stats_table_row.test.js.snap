exports[`TableRow should match the snapshot 1`] = `
<tr
  className="inner-text">
  <td>
    Alice
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0
  </td>
  <td>
    0%
  </td>
</tr>
`;
